evm/chains/base.py

```diff
@@ -686,15 +687,26 @@ def validate_uncles(self, block: BaseBlock) -> None:
             uncle_vm_class = self.get_vm_class_for_block_number(uncle.block_number)
             uncle_vm_class.validate_uncle(block, uncle, uncle_parent)
 
-    def validate_chain(self, chain: Tuple[BlockHeader, ...]) -> None:
+    def validate_chain(self, chain: Tuple[BlockHeader, ...], seal_check_frequency: int = 1) -> None:
         parent = self.chaindb.get_block_header_by_hash(chain[0].parent_hash)
-        for header in chain:
+        all_indices = list(range(len(chain)))
+        if seal_check_frequency == 1:
+            headers_to_check_seal = all_indices
+        else:
+            headers_to_check_seal = random.sample(
+                all_indices, len(all_indices) // seal_check_frequency)
+
+        for i, header in enumerate(chain):
             if header.parent_hash != parent.hash:
                 raise ValidationError(
                     "Invalid header chain; {} has parent {}, but expected {}".format(
                         header, header.parent_hash, parent.hash))
             vm_class = self.get_vm_class_for_block_number(header.block_number)
-            vm_class.validate_header(header, parent)
+            if i in headers_to_check_seal:
+                check_seal = True
+            else:
+                check_seal = False
```

Comment:
Maybe just embed the call to `vm_class.validate_header(header, parent, check_seal=True)` here and get rid of the `check_seal` variable?

Reply:
Well, `validate_header()` does more than checking the seal, and we still want to perform those for every header -- it's only the seal check that we want to skip sometimes

Reply:
Oh, duh, now I understand what you mean. Will do that